drivers/clock_control/clock_stm32_ll_h7.c

```diff
@@ -147,6 +152,7 @@ static inline int stm32_clock_control_off(struct device *dev,
 	default:
 		return -ENOTSUP;
```

Comment:
This needs to be something like:
```
    rc = -ENOTSOP;;
   goto out;
```
otherwise you're returning while `hsem_lock` is still held.

Please check for this problem elsewhere too.

Reply:
Nice catch, thanks. Done